Signal/src/AppDelegate.m

```diff
@@ -594,6 +599,17 @@ - (void)applicationDidBecomeActive:(UIApplication *)application {
     DDLogInfo(@"%@ applicationDidBecomeActive completed.", self.logTag);
 }
 
+- (void)enableBackgroundRefreshIfNecessary
+{
+    if (OWS2FAManager.sharedManager.is2FAEnabled && [TSAccountManager isRegistered]) {
```

Comment:
So this is safe to before "app is ready".

Reply:
I took care to only run it in other blocks protected by app-readiness, but being redundant here is a good idea in case it would get moved. Updated!